src/Validation/MailValidator.php

```diff
@@ -13,14 +14,11 @@
  */
 class MailValidator implements ScalarValueValidator {
 
-	private $testWithMX;
+	private $domainValidator;
 	private $lastViolation;
 
-	const TEST_WITH_MX = true;
-	const TEST_WITHOUT_MX = false;
-
-	public function __construct( bool $testWithMX ) {
-		$this->testWithMX = $testWithMX;
+	public function __construct( DomainNameValidator $tldValidator ) {
```

Comment:
Yeah, I forgot to rename this var. Originally I thought this stuff only did TLD validation.